File: 22-10-2023_sub_arrays_with_equal_number_of_occurences.py
class Solution:
    def sameOccurrence(self, arr, x, y):
        dict1 = {0: 1}
        count = 0
        output = 0
        for num in arr:
            if num == x:
                count += 1
            elif num == y:
                count -= 1
            if count in dict1:
                output += dict1[count]
            dict1[count] = dict1.get(count, 0) + 1
        return output

File: 22-10-2023_sub_arrays_with_equal_number_of_occurences.java
class Solution {

    static int sameOccurrence(int arr[], int x, int y) {
        Map<Integer, Integer> dict = new HashMap();
        int count = 0;
        int output = 0;
        dict.put(0, 1);
        for (int num: arr) {
            if (num == x) {
                count++;
            } else if (num == y) {
                count--;
            }
            if (dict.containsKey(count)) {
                output += dict.get(count);
            }
            dict.put(count, dict.getOrDefault(count, 0) + 1);
        }
        return output;
    }
}
